product-scenarios/1-manage-public-partner-private-apis/1.4-delete-an-existing-api/1.4.1-delete-an-existing-api/src/test/java/org/wso2/am/scenario/tests/delete/existing/api/DeleteAPINegativeTestCases.java

```diff
@@ -0,0 +1,228 @@
+    private static final String API_SUBSCRIBER_USERNAME = "DeleteAPISubscriberNeg";
+    private static final String API_SUBSCRIBER_PW = "DeleteAPISubscriberNeg";
+    private static final String API_PUBLISHER_USERNAME = "DeleteAPIPublisherNeg";
+    private static final String API_PUBLISHER_PW = "DeleteAPIPublisherNeg";
+    private static final String API_NAME_PREFIX = "DeleteAPINeg_";
+    private static final String API_VERSION = "1.0.0";
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws APIManagerIntegrationTestException, APIManagementException, RemoteException,
+            UserAdminUserAdminException {
+        createUserWithPublisherAndCreatorRole(API_CREATOR_PUBLISHER_USERNAME, API_CREATOR_PUBLISHER_PW, ADMIN_USERNAME,
+                ADMIN_PW);
+        apiPublisher = new APIPublisherRestClient(publisherURL);
+        apiPublisher.login(API_CREATOR_PUBLISHER_USERNAME, API_CREATOR_PUBLISHER_PW);
+        createUserWithSubscriberRole(API_SUBSCRIBER_USERNAME, API_SUBSCRIBER_PW, ADMIN_USERNAME, ADMIN_PW);
+        apiStore = new APIStoreRestClient(storeURL);
+        apiStore.login(API_SUBSCRIBER_USERNAME, API_SUBSCRIBER_PW);
+    }
+
+    @Test(description = "1.4.1.4", dataProvider = "DeleteAPIAfterSubscribingDataProvider",
+            dataProviderClass = ScenarioDataProvider.class)
+    public void testDeleteAPIWithSubscription(APILifeCycleState state) throws Exception {
+        String name = API_NAME_PREFIX + "subAppNeg_" + state.toString();
+        createAPIAndSubscribe(name, name);
+        if (state.equals(APILifeCycleState.BLOCKED) || state.equals(APILifeCycleState.DEPRECATED)) {
+            changeApiState(name, state);
+        }
+        checkDeleteAPI(name, "Cannot remove the API as active ");
+        verifyAPIAvailableInPublisher(name);
+//        check availability in store for only PUBLISHED apis
+        if (state.equals(APILifeCycleState.PUBLISHED)) {
+            verifyAPIAvailableInStore(name);
+        }
+    }
+
+    @Test(description = "1.4.1.5")
+    public void testDeleteAPIByUnauthorizedUser() throws Exception {
+        String name = API_NAME_PREFIX + "unauthUser";
+        createApi(name);
+        createUserWithPublisherRole(API_PUBLISHER_USERNAME, API_PUBLISHER_PW, ADMIN_USERNAME, ADMIN_PW);
+        apiPublisher.login(API_PUBLISHER_USERNAME, API_PUBLISHER_PW);
+        checkDeleteAPI(name,
+                "does not have the required permission: /permission/admin/manage/api/create");
+        apiPublisher.login(API_CREATOR_PUBLISHER_USERNAME, API_CREATOR_PUBLISHER_PW);
+//        API is in CREATED state therefor only check availability in publisher
+        verifyAPIAvailableInPublisher(name);
+    }
+
+    @Test(description = "1.4.1.6")
+    public void testDeleteNonExistingAPI() throws Exception {
+        checkDeleteAPI(API_NAME_PREFIX + "nonExist", "Unable to find the API");
+    }
+
+    private void createAPIAndSubscribe(String apiName, String applicationName) throws Exception{
```

Comment:
Add doc comments for methods. Fix at other places too

Reply:
Fixed with #5210